src/components/Reports/ReportsBox/ReportsBox.js

```diff
@@ -1,122 +1,77 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import { InjectAppServices } from '../../../services/pure-di';
 import { FormattedMessage } from 'react-intl';
 import { FormattedDateRangeText } from '../../shared/FormattedDateRangeText/FormattedDateRangeText';
 
-class ReportsBox extends React.Component {
-  /**
-   * @param { Object } props - props
-   * @param { import('../../../services/pure-di').AppServices } props.dependencies
-   */
-  constructor({ dependencies: { datahubClient } }) {
-    super();
+/**
+ * @param { Object } props - props
+ * @param { import('../../../services/pure-di').AppServices } props.dependencies
+ */
+const ReportsBox = ({
+  domainName,
+  dateFrom,
+  dateTo,
+  withEmail,
+  withoutEmail,
+  today,
+  dependencies: { datahubClient },
+}) => {
+  const [visits, setVisits] = useState(null);
 
-    this.datahubClient = datahubClient;
-
-    this.state = {
-      visits: null,
+  useEffect(() => {
+    const fetchVisitsByPeriod = async (domainName, dateFrom, dateTo) => {
+      let asyncRequest = datahubClient.getTotalVisitsOfPeriod({
```

Comment:
what do you think to use `const asyncRequest =  await datahubClient.getTotalVisitsOfPeriod({` ? no need to put in null after